props-core/src/main/java/sh/props/Registry.java

```diff
@@ -92,8 +105,18 @@ public void sendUpdate(String key, @Nullable String value, @Nullable Layer layer
    * @throws IllegalArgumentException if a previously bound prop is passed
    */
   public <T, PropT extends AbstractProp<T>> PropT bind(PropT prop) {
+    final String key = prop.key();
+
+    // notify all layers that the specified prop was bound
+    // but do not wait for the key to be loaded
+    for (Layer layer : layers) {
+      @SuppressWarnings({"FutureReturnValueIgnored", "UnusedVariable"})
+      var future = layer.registerKey(key);
```

Comment:
Registers the key after binding, but does not wait for values to be retrieved/updated. This is consistent with how `bind` works (it will receive an updated value, async)